### ISSUE TITLE
FR: Bulldog Magazine Filter Blocks 

### ISSUE THREAD

**Issue by bfvolvo** [aemsites/vg-macktrucks-com#41]
### Context/Description

Dependent on https://github.com/aemsites/vg-macktrucks-com/issues/106 
Related to https://github.com/aemsites/volvotrucks-na/issues/27 for repurposing code

Update Nov 4
Based on ticket https://github.com/hlxsites/vg-macktrucks-com/issues/808, the same logic for filtering and categories in search can be applied however visuals need to be developed on the FE to reflect the MT designs, this work was started in ticket https://github.com/hlxsites/vg-macktrucks-com/issues/602 by @jonatan-lledo-netcentric. 

This is an extension of ticket 808, which focused primarily on a single component design, namely a multi-select drop down. From a prior refinement meeting where we discussed ticket 808 it was mentioned that styling a multi-select drop down would be difficult. This ticket extends the design thinking to be more page inclusive and format inclusive for mobile. This approach includes some additional UI such as sorting and label buttons to enhance the experience and allow users to quickly find articles.

Context/Description
As a user visiting the Bulldog Magazine website, I want to search/filter/sort articles based on categories, topics, and truck models so that I can easily find relevant articles that interest me.

### Acceptance Criteria

**User AC:**

- [x]  All filtering allows for multiple options to be selected so the user can select multiple fields in the filter menu. For example, a user can select multiple truck models and topics at the same time.
- [x]  When a user selects a filter or filters, indication should be provided with [Chips](https://www.volvogroup.com/experiencesystem/en/#/brand/615bfcc428c818001105b4e3/61a0de32a7047974164f01dd/62308e2a37a675408d85d7fb) (Styled to Mack Trucks) so that user see what filters have been applied. These tags should be dismissible, thus removing a filter option or all by clicking "Clear all"
- [x]  When a user selects a filter, the article content should dynamically update to show only those articles that match all selected criteria without any page reloads.
- [x]  The filter menu should have a scroll bar if the user's screen is not large enough to show the entire list.
- [x] A No Options Available error message shows if there is no return on the combination of chips (visible in the Figma design)

**Desktop/tablet:**
- [x] On desktop/tablet, the categories are accessible from a filters icon label and the categories are displayed in a menu, three column format with a category header and checkbox options below each category.
- [x] To provide indication to end users of which filters have been selected a chips group would populate below the filters and area, where a user can see what filters are applied during a session and instantly remove one or clear all.
- [x] Maximum of 7 chips can be shown, beyond that an 8th chip appears with indication of "X more". Click the "X more" tag reveals remaining filters applied. X is the number of filters applied after 7.
- [x] Clicking clear all clears all filters.

**Mobile:**
- [x] On mobile devices, clicking the filter menu launches an overlay menu that fills 90% of the screen from top to bottom
- [x] The overlay menu contains a header where there is a close button "X" icon and below that is an accordion stack showing categories, The user can expand and collapse a category to see its filter options in an [exclusive accordion](https://developer.chrome.com/docs/css-ui/exclusive-accordion#the_exclusive_accordion) manner. From there the user can multi-select multiple filter options, upon doing this a number increment appears next to the accordion category name in the following format "(X applied)" where X is the number of selected filter options for that category.
- [ ] On mobile, When the user selects "Close" and/or "Apply" from the overlay menu, the overlay menu will automatically close, returning the user to the previous state. "Close" button closes the menu and no filters are applied, "Apply" button applies the filters and closes the menu.
- [x] On mobile, if the user selects multiple categories, after the users close out of the filter bar it will show the number of categories selected. (example: 3 Applied)
- [x] Clicking the "Clear" button clears all previously selected filters and allows user to reapply a new set or close out.
- [ ] Similar to desktop a user can sort by "Most Recent" and "Alphabetical" by drop down.

**Categories and Filter Options:**

 **1. All Categories** 
       - Customer Profiles 
       - In the Know
       - Product Updates 
       - Technology at Work
       - Features 
       - In the News 
       - Viewpoint
       - Calendar 

    **2. All Topics** 
       - Topic options are TBD @WendyKruger 
    
  **3. All Trucks** 
       - Anthem
       - Granite
       - LR
       - LR Electric
       - MD 
       - MD Electric
       - Pinnacle
       - TerraPro

**Editor AC**

- [ ] Content editors can add new filter tags, edit existing ones, and remove tags as needed.

### Open Questions:
Is it possible for content editors to manage the tags using a solution that's in place for Volvo? https://main--vg-volvotrucks-us--hlxsites.aem.page/tools/tagger/index.html

### Figma Designs
Dev Ready:
https://www.figma.com/design/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?node-id=2210-65358&m=dev&focus-id=17058-37745

Desktop Prototype - https://www.figma.com/proto/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?page-id=2210%3A65358&node-id=17043-4563&node-type=frame&viewport=-2671%2C-2358%2C0.33&t=pMQiiNAk2rtGgOJm-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=17023%3A18754&show-proto-sidebar=1
<img width="541" alt="{9817B6A7-91D7-478C-969F-95B39F93B6B8}" src="https://github.com/user-attachments/assets/13067d8e-fd65-4205-8156-13271e98820f">
<img width="541" alt="{ED64CD68-3F13-43C8-AE13-BD43F6645B0C}" src="https://github.com/user-attachments/assets/c524f1fd-09cc-4381-9323-ff6e8e0a0fd1">

Mobile Prototype - https://www.figma.com/proto/oNwe2jHIICPMIe551EY62l/Mack-Reskin---Full-Project?page-id=2210%3A65358&node-id=17070-90430&node-type=frame&viewport=-2671%2C-2358%2C0.33&t=pMQiiNAk2rtGgOJm-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=17023%3A18754&show-proto-sidebar=1

<img width="238" alt="{6342EE7B-343D-4B8B-95E9-14F1B280D556}" src="https://github.com/user-attachments/assets/7a87a76e-fa6e-41ca-aeb4-68a29ce73c9a">

### Out of scope

Search functionality for magazine will be handled in a [separate ticket](https://github.com/hlxsites/vg-macktrucks-com/issues/766)
- [ ] aemsites/vg-macktrucks-com#39
- [ ] Users can search articles due to the functionality of #106

### Technical details

We want to avoid what is currently happening on production for the All Truck models example where if an article is tagged with multiple product tags that it creates a separate tag line as seen here: https://www.macktrucks.com/magazine/explore-articles

**Comment by mollyaljenkins** [comment]
Depending on #116

**Comment by VladMano** [comment]
Testing completed, the following issues have been found:

- When selecting a filter, the articles do not dynamically update, as the page only updates after closing the filter page.
- On picking an invalid filter combination, you receive "Magazine error title, Magazine error" text instead of "No Options Available" error.
- The categories are not correctly ordered as they show up in the following order: 1.Truck / 2.Topic / 3.Category
- The subcategories are ordered incorrectly and one is missing: "Calendar" from "All Categories"
- In the "Truck" Category, all of the subcategories start with "Mack" in front, which is different from what is showing in the AC.
- We need clarification for the "All Topics" cateogry, as they are TBD and we do not know which are the correct ones.


Desktop:
- When picking more than 5 chips, the other ones are displayed under the first row and they are clipped by the article themselves (As per the AC, 7 chips should show up and there should be a "X more" tag that reveals the remaining filters applied. See screensho attached ![image](https://github.com/user-attachments/assets/cd6bcc08-7d26-44ab-91f8-e40674c39c32)

Mobile:
- No "Most Recent" or "Alphabetical" in the drop down.
- When pressing the "Clear" button, you are not taken back to the article page automatically, as you have to press "Apply" afterwards.
- Missing "X" button from the filters page

**Comment by silviulungu** [comment]
Testing completed. As discussed with @bfvolvo, the "Close" button works as intended even if is not matching with the AC specified inside the ticket and the drop down for sorting will be tackled in another ticket. The two AC boxes will remain unchecked as a reminder of what was decided. Besides these two observations, everything works accordingly.

**Comment by santi-homps** [comment]
Close btn functionality for mobile applies the desktop behaviour where whenever the filters are closed, the search is done and the articles updated since if not, the state of the on/off filters is too hard to manage. 
Sorting will be tackled in issue #39

**Comment by BeckyMedlin** [comment]
@santi-homps Please close this ticket; it's deployed. Thanks!
